src/firefly/js/metaConvert/vo/DatalinkProducts.js

```diff
@@ -114,6 +109,99 @@ export async function getDatalinkRelatedGridProduct({dlTableUrl, activateParams,
     }
 }
 
+
+export async function getObsCoreRelatedDataProductByFilter(table, row, threeColorOps, highlightPlotId, activateParams, options) {
+
+    dispatchUpdateDataProducts(activateParams.dpId, dpdtWorkingMessage('Loading data products...', 'working'));
+    const preferCutout= getPreferCutout(options.dataProductsComponentKey,table?.tbl_id);
+
+    try {
+        const filteredTbl= await fetchRelatedColsFilteredTable(table,row);
+        const dlTableAry= await fetchAllDatalinkTables(filteredTbl,options.datalinkTblRequestOptions);
+        const {processedGrid,gridDlData, errMsg}=
+            convertDLTableAryToImageGrid(dlTableAry,filteredTbl,preferCutout,activateParams,options);
+        if (errMsg) return dpdtSimpleMsg(errMsg);
+
+
+        const requestAry= processedGrid.map( (g) => g.request);
+        requestAry.forEach((r, idx) => r.setPlotId(r.getPlotId() + '-related_grid-' + idx));
+        const threeColorReqAry= (threeColorOps && requestAry.length>1) &&
+            make3ColorRequestAry(requestAry,threeColorOps,table.tbl_id);
+        const onActivePvChanged= getOnActivePvChanged(getObsCoreAccessURL(filteredTbl,0),processedGrid,activateParams,options);
+        const {imageViewerId} = activateParams;
+        const activate = createRelatedGridImagesActivate({requestAry, threeColorReqAry, imageViewerId,
+            tbl_id:table.tbl_id, onActivePvChanged, highlightPlotId });
+        const extraction = createSingleImageExtraction(requestAry);
+
+        const item= dpdtImage({name:'image grid', activate, extraction,
+            dlData: gridDlData,
+            enableCutout:preferCutout,
+            menu: [],
+            menuKey:'image-grid-0',});
+
+        return item;
+    }
+    catch (reason) {
+        logger.error(`can't filter table: ${reason}`, reason);
+        return dpdtSimpleMsg('can\'t filter table');
+    }
+
+}
+
+
+async function fetchRelatedColsFilteredTable(table,row) {
+    const s= getMetaEntry(table,'tbl.relatedCols');
+    const colNameAry= s.split(',').map( (c) => c.trim());
+    const values= colNameAry.map( (n) => getCellValue(table,row,n));
+    const filters= colNameAry.reduce ( (str,cName,idx) =>
+        `${str}${!idx?'':' AND '}"${cName}" = '${values[idx]}'`, '');
+    const request = cloneRequest(table.request, { startIdx : 0, pageSize : MAX_ROW, filters});
+
+    try {
+        return await doFetchTable(request);
+    }
+    catch (reason) {
+        console.log('could not filter tabel', reason);
```

Comment:
shouldn't be logged to console, I think
```suggestion
        logger.error('could not filter tabel', reason);
```